src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java

```diff
@@ -178,7 +187,14 @@ public Object peekCause() {
     @Override
     public StackFrame pushCauseFrame() {
         enforceMainThread();
-        CauseStackFrameImpl frame = new CauseStackFrameImpl(this.min_depth);
+        CauseStackFrameImpl frame;
+        if (this.framePool.isEmpty()) {
+            frame = new CauseStackFrameImpl();
+        } else {
+            frame = this.framePool.pop();
```

Comment:
How does this interact with the debugging of stack frames with the exception built in?